.gitattributes

```diff
@@ -0,0 +1,15 @@
+# Lock batch file ending
+*.bat text eol=crlf
+
+# Checkout as OS specific line endings, check in as LF
+* text eol=lf
+
+# Unset formating for binary files
+*.gif binary
+*.jar binary
+*.jpg binary
+*.png binary
+*.tar.gz binary
+*.tgz binary
+*.ttf binary
+*.zip binary
```

Comment:
Based on my reading, each of these lines tells Git to treat a specific file extension -- `.gif`, `.jar`, _etc._ -- as binary.

These lines override the previous rule (setting everything to text) for the extensions listed.